Bump ASM opcode version

<!--- Please follow the instructions below. We receive dozens of issues every week, -->
<!--- so to stay productive, we will close issues that don't provide enough information. -->

<!--- Provide a brief summary of the issue in the title above -->
If project has an `record` class in it the `com.bmuschko.gradle.docker.internal.MainClassFinder` is unable to find the main class.

### Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
Successfully find the main class.

### Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
Throws an exception:

```
Caused by: java.lang.UnsupportedOperationException: Records requires ASM8
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassVisitor.visit(ClassVisitor.java:112)
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.accept(ClassReader.java:546)
	at com.bmuschko.gradle.docker.shaded.org.objectweb.asm.ClassReader.accept(ClassReader.java:401)
	at com.bmuschko.gradle.docker.internal.MainClassFinder.createClassDescriptor(MainClassFinder.groovy:163)
	at com.bmuschko.gradle.docker.internal.MainClassFinder.doWithMainClasses(MainClassFinder.groovy:132)
	at com.bmuschko.gradle.docker.internal.MainClassFinder.findSingleMainClass(MainClassFinder.groovy:27)
	at com.bmuschko.gradle.docker.DockerSpringBootApplicationPlugin.findMainClassName(DockerSpringBootApplicationPlugin.groovy:32)
	at com.bmuschko.gradle.docker.DockerConventionJvmApplicationPlugin.getApplicationMainClassName(DockerConventionJvmApplicationPlugin.groovy:205)
	at com.bmuschko.gradle.docker.DockerConventionJvmApplicationPlugin.access$1(DockerConventionJvmApplicationPlugin.groovy)
	at com.bmuschko.gradle.docker.DockerConventionJvmApplicationPlugin$2$7.call(DockerConventionJvmApplicationPlugin.groovy:131)
	at com.bmuschko.gradle.docker.DockerConventionJvmApplicationPlugin$2$7.call(DockerConventionJvmApplicationPlugin.groovy)
	at org.gradle.api.internal.provider.DefaultProvider.calculateOwnValue(DefaultProvider.java:66)
```
### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
Unable to use `gradle-docker-plugin` if have an `record` classes in the source code.

### Steps to Reproduce (for bugs)
<!--- Please provide one of the below: -->
<!--- Create a functionalTest within this project which reproduces the problem on a new branch or PR. -->
<!--- Provide a self-contained example project (as an attached archive or a Github project). -->
<!--- In the rare cases where this is infeasible, we will also accept a detailed set of instructions. -->
In order to reproduce put a `record` in the project source code and execute `dockerCreateDockerfile` task.

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

    plugin-version: 7.0.0
    gradle-version: 6.8.3
    docker-version: 20.10.6
    platform: Ubuntu 18.04.5

--- a/src/main/groovy/com/bmuschko/gradle/docker/internal/MainClassFinder.groovy
+++ b/src/main/groovy/com/bmuschko/gradle/docker/internal/MainClassFinder.groovy
@@ -195,7 +195,7 @@ class MainClassFinder {
         private boolean mainMethodFound
 
         ClassDescriptor() {
-            super(Opcodes.ASM7)
+            super(Opcodes.ASM9)
         }
 
         @Override

